pkg/cloudprovider/controller/nodecontroller.go

```diff
@@ -80,6 +80,49 @@ func NewNodeController(
 	}
 }
 
+// Generates num pod CIDRs that could be assigned to nodes.
+func generateCIDRs(num int) util.StringSet {
+	res := util.NewStringSet()
+	for i := 0; i < num; i++ {
+		// TODO: Make the CIDRs configurable.
+		res.Insert(fmt.Sprintf("10.244.%v.0/24", i))
+	}
+	return res
+}
+
+// For each node from newNodes, finds its current spec in registeredNodes.
+// If it is not there, it gets a new valid CIDR assigned.
+func reconcilePodCIDRs(newNodes, registeredNodes *api.NodeList) *api.NodeList {
+	registeredCIDRs := make(map[string]string)
+	availableCIDRs := generateCIDRs(len(newNodes.Items) + len(registeredNodes.Items))
+	for _, node := range registeredNodes.Items {
+		registeredCIDRs[node.Name] = node.Spec.PodCIDR
+		availableCIDRs.Delete(node.Spec.PodCIDR)
+	}
+	for i, node := range newNodes.Items {
+		podCIDR, registered := registeredCIDRs[node.Name]
+		if !registered {
+			podCIDR, _ = availableCIDRs.PopAny()
+		}
+		newNodes.Items[i].Spec.PodCIDR = podCIDR
+	}
+	return newNodes
+}
+
+func (s *NodeController) configureNodeCIDR(node *api.Node) {
+	var err error
+	if instances, ok := s.cloud.Instances(); ok {
```

Comment:
The go style guide would say that you should invert the logic here so that only the exceptional case is indented. I think that would also make the error flow easier to follow:

instances, ok := s.cloud.Instances()
if !ok {
	glog.Errorf("Error configuring node %s: CloudProvider does not support Instances()", node.Name)
	return
}
if err := instances.Configure(node.Name, &node.Spec); err != nil {
	glog.Errorf("Error configuring node %s: %s", node.Name, err)
	node.Spec.PodCIDR = ""
	return
}